File: ibkr_show_backend/app/schemas/admin_ibkr.py
from pydantic import BaseModel


class IBKRFlexSettingsResponse(BaseModel):
    query_id: str
    flex_token_masked: str
    has_flex_token: bool
    config_file: str


class IBKRFlexSettingsUpdateRequest(BaseModel):
    query_id: str | None = None
    flex_token: str | None = None


class IBKRFlexSettingsMutationResponse(BaseModel):
    settings: IBKRFlexSettingsResponse
    message: str


class IBKRFlexTestResponse(BaseModel):
    success: bool
    query_id: str
    reference_code: str | None = None
    message: str | None = None


class IBKRImportResponse(BaseModel):
    success: bool
    filename: str
    result: dict
    message: str

File: ibkr_show_frontend/src/types/adminIbkr.ts
export type IbkrFlexSettings = {
  query_id: string
  flex_token_masked: string
  has_flex_token: boolean
  config_file: string
}

export type IbkrFlexSettingsPayload = {
  query_id?: string
  flex_token?: string
}

export type IbkrFlexSettingsMutationResponse = {
  settings: IbkrFlexSettings
  message: string
}

export type IbkrFlexTestResponse = {
  success: boolean
  query_id: string
  reference_code: string | null
  message: string | null
}

export type IbkrImportIndexResult = {
  index: string
  upserted: number
}

export type IbkrImportResponse = {
  success: boolean
  filename: string
  result: Record<string, IbkrImportIndexResult>
  message: string
}

